### ISSUE TITLE
Could you please explain why the training parameter for Batch Norm should be set to True during testing?

### ISSUE THREAD

**Issue by WKQ9411** [khoinguyenucd/LearningToCompare-Tensorflow#4]
```
def test(feature_encoder, relation_network, samples, sample_labels, batches, batch_labels):
    sample_features = feature_encoder(samples, True)
    sample_features = tf.reshape(sample_features, (CLASS_NUM, SAMPLE_NUM_PER_CLASS, 19, 19, FEATURE_DIM))
    sample_features = tf.math.reduce_sum(sample_features, 1)
    batch_features = feature_encoder(batches, True)

    sample_features_ext = tf.repeat(tf.expand_dims(sample_features, 0), 5 * CLASS_NUM, axis=0)
    batch_features_ext = tf.repeat(tf.expand_dims(batch_features, 0), CLASS_NUM, axis=0)
    batch_features_ext = tf.transpose(batch_features_ext, (1, 0, 2, 3, 4))
    relation_pairs = tf.reshape(tf.concat([sample_features_ext, batch_features_ext], axis=4),
                                (-1, 19, 19, FEATURE_DIM * 2))
    relations = tf.reshape(relation_network(relation_pairs, True), (-1, CLASS_NUM))

    labels = tf.math.argmax(tf.squeeze(batch_labels), 1)
    predict_labels = tf.math.argmax(relations, 1)

    rewards = [1.0 if predict_labels[i] == labels[i] else 0.0 for i in range(25)]

    return np.sum(rewards) / 25.0
```

**Comment by khoiucd** [comment]
When it is set to True, the Batch Norm will use the statistics from the samples in the episode (including query and support samples), instead of taking the learned statistics. This allows information to be exchanged among the samples, leading to better acuraccy. The setting is in fact transductive few-shot learning, instead of inductive few-shot learning.

**Comment by WKQ9411** [comment]
Thank you very much for your detailed answer.
